plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/MethodEmbedding.kt

```diff
@@ -60,11 +60,18 @@ class UserMethodEmbedding(
             override val method: MethodEmbedding = this@UserMethodEmbedding
 
             override val nameMangler = NoopNameMangler
+            override val scopeDepth: Int
+                get() = 0
+
             override fun getLambdaOrNull(name: Name): SubstitutionLambda? = null
         }
 
         body = symbol.fir.body?.let {
-            val stmtCtx = StmtConverter(methodCtx, SeqnBuilder(), NoopResultTrackerFactory)
+            val stmtCtx = StmtConverter(methodCtx, SeqnBuilder(), NoopResultTrackerFactory, scopeDepth = 0)
+            methodCtx.method.params.forEach { varEmb ->
+                // TODO: is there a better way to do that?
+                stmtCtx.addScopedName((varEmb.name as LocalName).name)
+            }
```

Comment:
One way to approach this is to make sure we never use `VariableEmbedding` for fields, and then you can use `LocalName` as the type of `name` there, I think.

Reply:
do you think it makes sense doing that in this PR?

Reply:
Yes, it shouldn't be very big.